mtt/src/mtt_types.rs

```diff
@@ -118,3 +116,226 @@ impl MttL1EntryType {
         }
     }
 }
+
+// Bits 45:26
+const L2_ADDR_MASK: u64 = 0x3fff_fc00_0000;
+// Shifted right by 26, then 3-bit left shift to account for the 8-byte entry
+const L2_ADDR_SHIFT: u64 = 23;
+
+fn get_l2_entry(phys_addr: SupervisorPageAddr) -> MttL2EntryType {
+    let addr = phys_addr.bits();
+    let mtt_index = (addr & L2_ADDR_MASK) >> L2_ADDR_SHIFT;
```

Comment:
nit: addr_to_index can be a const function as it's open coded in at least two places.

Reply:
Ack.